### PR TITLE
Add usage documentation in README

### PR DESCRIPTION
**Pull request by eskatos**
This PR adds some usage documentation to the project's README file.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,6 +7,106 @@
 
 This is a desktop application acting as a Gradle Tooling API client.
 
+## Usage
+
+You can download the latest version of _Gradle Client_ for macos, Linux or Windows from the [Releases](https://github.com/gradle/gradle-client/releases) assets.
+
+Once installed on your system, simply launch the application.
```
Comment (tresat) [inline README.md:14]: 💭 Installed?  This makes it sound like a much more heavyweight application.  This is just download and run.

File: README.md [hunk 2]
```diff
@@ -7,6 +7,106 @@
 
 This is a desktop application acting as a Gradle Tooling API client.
 
+## Usage
+
+You can download the latest version of _Gradle Client_ for macos, Linux or Windows from the [Releases](https://github.com/gradle/gradle-client/releases) assets.
```
Comment (tresat) [inline README.md:12]: 💭 There aren't any releases here yet.

File: README.md [hunk 3]
```diff
@@ -7,6 +7,106 @@
...
+
+Once installed on your system, simply launch the application.
+
+### Adding a build
+
+The first screen shown by the _Gradle Client_ allows you to add local Gradle builds you want to interact with.
```
Comment (tresat) [inline README.md:18]:
```suggestion
The first screen shown by the _Gradle Client_ allows you to add local Gradle builds to explore.
```

File: README.md [hunk 4]
```diff
@@ -7,6 +7,106 @@
 
 This is a desktop application acting as a Gradle Tooling API client.
 
+## Usage
```
Comment (tresat) [inline README.md:10]: 💭 Maybe a sentence or two at the top that describes what this project is for.

File: README.md [hunk 5]
```diff
@@ -7,6 +7,106 @@
...
+**Declarative Schema** (_Only for builds using [Declarative Gradle](https://declarative.gradle.org/)_)
+
+* Informs about the schema of the available _Software Types_.
+* Fetches the [DeclarativeSchemaModel](https://github.com/gradle/gradle/blob/10b91d86d67226538bd721a2ee2aefb5233947d5/platforms/core-configuration/declarative-dsl-tooling-models/src/main/java/org/gradle/declarative/dsl/tooling/models/DeclarativeSchemaModel.java#L22) Tooling API model.
+* This action requires the completion of the [initialization phase](https://docs.gradle.org/current/userguide/build_lifecycle.html#sec:initialization) of the build.
+* No subprojects of the build are configured.
+
+**Declarative Documents** (_Only for builds using [Declarative Gradle](https://declarative.gradle.org/)_)
+
+* Informs about the declarative build definition and allows to trigger mutations.
```
Comment (tresat) [inline README.md:94]:
```suggestion
* Informs about the declarative build definition and allows triggering mutations.
```

**Review by tresat — APPROVED** [review]

**Comment by big-guy** [comment]
Can we include this in our declarative-gradle/declarative.gradle.org instructions somehow?

File: README.md (hunk 2 above)
Comment (eskatos) [reply README.md:12]: Right. I just published the first release, PTAL.

File: README.md (hunk 1 above)
Comment (eskatos) [reply README.md:14]: The binaries are respectively `dmg`, `deb` and `msi` files, so they need to be installed.

**Comment by eskatos** [comment]
> Can we include this in our declarative-gradle/declarative.gradle.org instructions somehow?

Yes. Let's get that merged first.

File: README.md (hunk 1 above)
Comment (tresat) [reply README.md:14]: Oh, ok.  I just built and ran the project and didn't notice that.

File: README.md (hunk 4 above)
Comment (tresat) [reply README.md:10]: "This is a desktop application acting as a Gradle Tooling API client."

I don't think this is very descriptive or helpful.  What does it demonstrate?  Why does it use the TAPI?  Things like that.

Comment (eskatos) [reply README.md:13]: I added some words, PTAL

Comment (tresat) [reply README.md:13]: Looks good!

**Pull request merged**